walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,189 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+
 
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.*; 
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+
+    //define variables
+    int dayStart = TimeRange.START_OF_DAY;
+    int dayEnd = TimeRange.END_OF_DAY;
+    int numberOfAttendees = request.getAttendees().size();
+    int numberOfEvents = events.size();
+    long meetingDuration = request.getDuration();
+
+    //optionsForNoAttendees
+    if(request.getAttendees().isEmpty())
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+
+    //noOptionsForTooLongOfARequest
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration())
```

Comment:
Similar couple of notes as above: I would either get rid of the comment or make it more readable, you might also use `meetingDuration` since you defined it above, and I would add `{ }` curly braces to your `if` statement.